pkg/envinfo/url.go

```diff
@@ -191,7 +191,11 @@ func (ei *EnvInfo) ValidateURL(url localConfigProvider.LocalURL) error {
 		}
 		for _, endpoint := range component.Container.Endpoints {
 			if endpoint.Name == url.Name && !ei.updateURL {
-				return fmt.Errorf("url %v already exist in devfile endpoint entry under container %v", url.Name, component.Name)
+				return fmt.Errorf("url %v already exists in devfile endpoint entry under container %q", url.Name, component.Name)
+			}
+			// Check for a duplicate port entry
+			if endpoint.TargetPort == url.Port {
```

Comment:
I'm not sure to understand when updateURL is set to true. Do you think it could be needed to test its value here also?

Reply:
I think it is set to true when you do not pass a URL name, and an invalid endpoint exists in the devfile. I'm not sure why it is as such, but that is my understanding.
https://github.com/redhat-developer/odo/blob/b75fa552ceb3daa2d37b4d931aca0f9b4686e86d/pkg/envinfo/url.go#L117-L131